### PR TITLE
new module syntrax

### PR DESCRIPTION
**Pull request by btwokr**
new module syntrax created; new syntrax generator added that uses jsyntrax API

### CODE REVIEW

File: syntrax/src/main/java/org/asciidoctor/diagram/syntrax/Syntrax.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        String[] options = new String[0];
+        String optionString = request.headers.getValue(HTTPHeader.OPTIONS);
+        if (optionString != null) {
+            options = optionString.split(" ");
+        }
+
+        InputArguments iArgs;
+        try {
+            iArgs = new InputArguments(options);
+        } catch (Exception e) {
```
Comment (inponomarev) [inline syntrax/src/main/java/org/asciidoctor/diagram/syntrax/Syntrax.java]: The correct way to handle: 
```java
catch (ParseException e) {
            throw new IOException(e);
        }
```

Thus we won't loose information about the call stack. And "when parsing input arguments" message does not add any significant information, as this all thing is buried deep in internals of Asciidoctor file processing

Comment (inponomarev) [reply syntrax/src/main/java/org/asciidoctor/diagram/syntrax/Syntrax.java]: And of course, we always must catch the most specific exception

File: syntrax/src/main/java/org/asciidoctor/diagram/syntrax/Syntrax.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @Override
+    public ResponseData generate(Request request) throws IOException {
+        MimeType format = request.headers.getValue(HTTPHeader.ACCEPT);
+        if (format == null) {
+            format = DEFAULT_OUTPUT_FORMAT;
+        }
+
+        if (!format.equals(MimeType.SVG)) {
```
Comment (inponomarev) [inline syntrax/src/main/java/org/asciidoctor/diagram/syntrax/Syntrax.java]: How about supporting PNG as well? Look how it's made in Main method of Jsyntrax, just one extra call is needed to convert SVG to PNG.

File: syntrax/build.gradle [hunk 3]
```diff
@@ -0,0 +1,12 @@
+repositories {
+    mavenLocal()
+    mavenCentral()
+    jcenter()
+}
+
+dependencies {
```
Comment (inponomarev) [inline syntrax/build.gradle:6]: nit: usually dependencies are sorted, implementation dependencies come fist, then come test dependencies. So please move the new dependency to line 9

**Review by inponomarev — CHANGES_REQUESTED** [review]
`META-INF.services` folder is missing, I think it won't work without it.
I see that there's no automated e2e test for this project, so we should test it manually:
1. `gradle install` this lib locally
2. In Celesta project, change the version in dependency to the local version
3. Try to build the docs with `mvn verify -pl celesta-documentation` and see if things are being done differently :-)

File: syntrax/build.gradle [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    implementation project(':server')
+    testImplementation project(':server')
+    testImplementation project(path: ':server', configuration: 'tests')
+    implementation 'org.atp-fivt:jsyntrax:1.31'
```
Comment (inponomarev) [inline syntrax/build.gradle]: 1.32 is deployed to public Maven repository with a public constructor for `InputArguments`. Please change to 1.32

File: syntrax/build.gradle (hunk 3 above)
Comment (btwokr) [reply syntrax/build.gradle:6]: done

File: syntrax/build.gradle (hunk 4 above)
Comment (btwokr) [reply syntrax/build.gradle]: done

File: syntrax/src/main/java/org/asciidoctor/diagram/syntrax/Syntrax.java (hunk 2 above)
Comment (btwokr) [reply syntrax/src/main/java/org/asciidoctor/diagram/syntrax/Syntrax.java]: new method and test added. 
SVGTranscoder.svg2Png(result) throws TranscoderException, which is caught and a new IOException exception is thrown

File: syntrax/src/main/java/org/asciidoctor/diagram/syntrax/Syntrax.java (hunk 1 above)
Comment (btwokr) [reply syntrax/src/main/java/org/asciidoctor/diagram/syntrax/Syntrax.java]: thanks for the feedback, now fixed!

**Comment by inponomarev** [comment]
Already merged to upstream, now closing this & syncing the `atp-mipt` fork

**Pull request closed without merging**